python/stack_of_tasks/ref_frame/__init__.py

```diff
@@ -1,91 +1,35 @@
-from __future__ import annotations
-
-from abc import abstractmethod
-
-from numpy.typing import NDArray
-from typing import Sequence, overload
 from typing_extensions import Protocol, runtime_checkable
 
 import numpy as np
 
 from tf.transformations import quaternion_matrix
 
-Transform = NDArray
-Jacobian = NDArray
+Transform = np.typing.NDArray
+Jacobian = np.typing.NDArray
 
 
 @runtime_checkable
 class HasTransform(Protocol):
     @property
-    @abstractmethod
     def T(self) -> Transform:
         pass
 
 
 @runtime_checkable
 class HasJacobian(Protocol):
     @property
-    @abstractmethod
     def J(self) -> Jacobian:
         pass
 
 
 @runtime_checkable
 class Transformable(Protocol):
-    @overload
     def transform(self, matrix: Transform) -> Transformable:
         pass
 
-    @overload
```

Comment:
Most of these overloaded methods are pure type-hits to show the reasonable combinations of arguments for the translate function.

Reply:
Doesn't my version has the same effect? Named optional arguments are _optional_ :wink: